Gemfile

```diff
@@ -1,6 +1,6 @@
 source 'https://rubygems.org'
```

Comment:
For code readability, it may be good to split the classes into separate files for both the spec files and actual codes being tested.